### PR TITLE
Bloom filter Join Step I: create benchmark

### PR DESCRIPTION
**Pull request by Lordworms**
## Which issue does this PR close?
part of #7955 

Closes #.

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
Thanks @Lordworms  -- I realize i am very behind on reviews in DataFusion

My first question on these benchmarks is are they measuring the right thing (namely are they dominated by the join time). Have you had a chance to run any profiling (flamegraphs, etc) to confirm these benchmarks are actually join dominated?

File: datafusion/core/benches/bloom_filter_join.rs [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+
+        // writing data
+        for row in data {
+            writeln!(writer, "{}", row.join(","))?;
+        }
+    }
+    Ok(temp_file)
+}
+
+fn create_tpch_q17_lineitem_csv() -> Result<NamedTempFile, std::io::Error> {
```
[the remark this answers is not in the document]
Comment (Lordworms) [reply datafusion/core/benches/bloom_filter_join.rs:62]: Sure, I'll refine this PR later this week

**Comment by alamb** [comment]
Marking as draft as I think this PR is no longer waiting on feedback. Please mark it as ready for review when it is ready for another look

**Comment by Lordworms** [comment]
for TPCH query 17, when we create 1000000 rows for lineitem and part table, the time spent on join is 50% (the other 80% of time spent on creating parquet files)
Screenshot 2024-08-25 at 1 37 15 PM

**Comment by Lordworms** [comment]
For the second case, 95% of the time spent on join
image

**Comment by Lordworms** [comment]
I think it worth a try to implement join predicate pushdown

**Comment by alamb** [comment]
I suggest we revive this PR as it seems to have gotten lost / not reviewed 😢

**Comment by Lordworms** [comment]
> I suggest we revive this PR as it seems to have gotten lost / not reviewed 😢

I am still working on the implementation of actual join_pushdown, I'll push a complete PR once it is done

**Comment by Lordworms** [comment]
> > I suggest we revive this PR as it seems to have gotten lost / not reviewed 😢
> 
I am still working on the implementation of actual "hash_join build side statistic pushdown", I'll push a complete PR once it is done

**Pull request closed without merging**